ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/TextEmbeddingModel.java

```diff
@@ -36,6 +46,48 @@ public ModelTensorOutput predict(String modelId, MLInput mlInput) throws Transla
         return new ModelTensorOutput(tensorOutputs);
     }
 
+    private boolean isAsymmetricModel(MLAlgoParams mlParams) {
+        if (mlParams instanceof AsymmetricTextEmbeddingParameters) {
+            // Check for the necessary prefixes in modelConfig
+            if (modelConfig == null
+                || ((TextEmbeddingModelConfig) modelConfig).getPassagePrefix() == null
+                    && ((TextEmbeddingModelConfig) modelConfig).getQueryPrefix() == null) {
+                throw new IllegalArgumentException(
+                    "When passing AsymmetricTextEmbeddingParameters, the model requires to be "
+                        + "registered with at least one of `query_prefix` or `passage_prefix`."
+                );
+            }
+            // Passed all checks
+            return true;
+        }
+
+        // no AsymmetricTextEmbeddingParameters passed, but the model is asymmetric.
+        if (modelConfig != null
+            && (((TextEmbeddingModelConfig) modelConfig).getPassagePrefix() != null
+                || ((TextEmbeddingModelConfig) modelConfig).getQueryPrefix() != null)) {
+            throw new IllegalArgumentException(
+                "The embedding model chosen is asymmetric. To use it, you must declare whether the input is of type `QUERY` or of type `PASSAGE`."
+            );
+        }
```

Comment:
Is this really necessary? Why not just return false in this condition. If the param is not asymmetric, just return false. Also this message is confusing. What if I passed a symmetric param with a "QUERY" by mistake, and this message doesn't help me to understand the problem?

Reply:
> Is this really necessary? Why not just return false in this condition. If the param is not asymmetric, just return false.

This exception will be thrown if users want to compute embeddings using an **asymmetric** model **without** stating what kind of data they are trying to embed. 

Running inference on an asymmetric model without the corresponding prefixes can lead to bad things happening. This is what this exception is avoiding. Returning `false` here would lead to OpenSearch silently accepting this and users wondering why the hell the supposedly superb models are performing bad. I don't think this is a good design choice.

> What if I passed a symmetric param with a "QUERY" by mistake, and this message doesn't help me to understand the problem?

I assume you mean what would happen if someone passed an `AsymmetricTextEmbeddingParameters` object, stating they want to embed a text of type `QUERY` while using a **symmetric** model (symmetric models don't take parameters)? This would lead to the exception in line 55. 

One could argue that this is exaggerated caution, that OpenSearch should accept such requests and simply embed the text without adding any prefixes. I thought about that and came to the conclusion, that it is better to fail here. This would avoid the error scenario that someone uploads and registers an **asymmetric** model and **forgets** to configure the prefixes. Trying to compute embeddings in this case would be then silently accepted by OpenSearch leading to degraded model performance.

> Also this message is confusing

Do you think this is still an issue? If so, could you suggest a change?